### ISSUE TITLE
setting variables escapes them

### ISSUE THREAD

**Issue by Riamse** [Keats/tera#196]
```jinja
{% set line = "&" %}
{% set line2 = line %}
{% set line3 = line2 %}
<span>{{line3}}</span>
```
It'll print `&amp;amp;amp;` which gets rendered as "&amp;amp;amp;". I do not think this is intentional.

**Comment by Riamse** [comment]
Seems that `{{line3|safe}}` gives us "&amp;amp;amp;", which is equivalent to `{{line2}}`, i.e. it undoes one level of escaping. You can't repeat safe pipes, though, it seems...

**Comment by Riamse** [comment]
Seems that piping into safe doesn't actually undo escaping, it just prevents escaping from happening https://github.com/Keats/tera/blob/bc2ade03b2a3f1e44e8fc73816841237807cc289/src/render.rs#L275 .

**Comment by Riamse** [comment]
Seems that the evaluation of Node::Set at https://github.com/Keats/tera/blob/bc2ade03b2a3f1e44e8fc73816841237807cc289/src/render.rs#L439 calls https://github.com/Keats/tera/blob/bc2ade03b2a3f1e44e8fc73816841237807cc289/src/render.rs#L330 which, when acting upon an ident, calls https://github.com/Keats/tera/blob/bc2ade03b2a3f1e44e8fc73816841237807cc289/src/render.rs#L244 which brings us back to the earlier link at https://github.com/Keats/tera/blob/bc2ade03b2a3f1e44e8fc73816841237807cc289/src/render.rs#L275 .

Calling `{% set x = y %}` shouldn't introduce the side effect of escaping `y`. I think a simple duct-tape fix would be to pipe a set assignment into the `safe` filter, but I think there might also be merit in having a separate function for this, or taking the `safe` pipe logic out of `eval_ident`.

What do you think?

**Comment by Keats** [comment]
> You can't repeat safe pipes, though, it seems...
Yeah right now the `safe` is applied after all the filters which is not what people expect I guess. I don't really remember why I did it that way though... Changing that would be a breaking change thought so that will be for 0.11

For the second case, I agree that the `set` shouldn't escape `y` in that case. I'm not too sure how to fix it, maybe applying an unescape before setting the variable? I'll think about it but I don't think we want to take the `safe` logic out of `eval_ident`, it looks like it would be too easy to forget escaping in some cases otherwise.

**Comment by Riamse** [comment]
That sounds fair. I don't think we'll have to go through the trouble of unescaping escaped content; just don't escape it in the first place. 

By the way, I got a patch working on my own computer that makes Node::Set add the `safe` filter to the rval when it evaluates. I can make a pull request if that's the desired solution.

**Comment by Keats** [comment]
Please send a PR! I will have a look.

**Comment by Keats** [comment]
The fix is in 0.10.9
